dubbo-common/src/main/java/org/apache/dubbo/common/timer/HashedWheelTimer.java

```diff
@@ -647,6 +679,26 @@ public void expire() {
                 return;
             }
 
+            // run timeout task at separate thread to avoid the worker thread blocking
+            try {
+                if (timer.taskExecutor == null) {
+                    // the global executor service should be gotten at each calling to avoid using invalid executor.
+                    GlobalResourcesRepository.getGlobalExecutorService().execute(this);
```

Comment:
`taskExecutor` must not null ? Whether to consider that are must required ？

Reply:
> `taskExecutor` must not null ? Whether to consider that are must required ？

Because we need to give the user the right to create thread pools to prevent creating too many thread pools

Reply:
> `taskExecutor` must not null ? Whether to consider that are must required ？

you might see the outdated codes.
```
private static final GlobalResourceInitializer<ExecutorService> DEFAULT_TIMER_TASK_EXECUTOR =
            new GlobalResourceInitializer<>(() ->
                Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors(),
                new NamedThreadFactory("HashedWheelTimerTask", true)),
            executor -> executor.shutdown());
```
the DEFAULT_TIMER_TASK_EXECUTOR  is only a GlobalResourceInitializer that the executor is only created when user decide to create the HashedWheelTimer by the default constructor instead of the new constructor which has added a new parameter taskExecutor:
```
    public HashedWheelTimer(
            ThreadFactory threadFactory,
            long tickDuration, TimeUnit unit, int ticksPerWheel,
            long maxPendingTimeouts, Executor taskExecutor) {
```

Reply:
> `taskExecutor` must not null ? Whether to consider that are must required ？

you might see the outdated PR codes.
```
private static final GlobalResourceInitializer<ExecutorService> DEFAULT_TIMER_TASK_EXECUTOR =
            new GlobalResourceInitializer<>(() ->
                Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors(),
                new NamedThreadFactory("HashedWheelTimerTask", true)),
            executor -> executor.shutdown());
```
the DEFAULT_TIMER_TASK_EXECUTOR  is only a GlobalResourceInitializer that the executor is only created when user decide to create the HashedWheelTimer by the default constructor instead of the new constructor which has added a new parameter taskExecutor:
```
    public HashedWheelTimer(
            ThreadFactory threadFactory,
            long tickDuration, TimeUnit unit, int ticksPerWheel,
            long maxPendingTimeouts, Executor taskExecutor) {
```